### PR TITLE
Holiday questionnaire: Implement the continuous placement condition

### PR DESCRIPTION
**Pull request by akheron**

### CODE REVIEW

File: service/src/main/kotlin/fi/espoo/evaka/holidayperiod/HolidayPeriodControllerCitizen.kt [hunk 1]
```diff
@@ -43,12 +49,38 @@ class HolidayPeriodControllerCitizen(private val accessControl: AccessControl) {
...
-    ): List<FixedPeriodQuestionnaire> {
+    ): List<FixedPeriodQuestionnaireWithChildren> {
         Audit.HolidayPeriodsList.log()
         accessControl.requirePermissionFor(user, Action.Global.READ_ACTIVE_HOLIDAY_QUESTIONNAIRES)
-        return db.connect { dbc -> dbc.read { listOfNotNull(it.getActiveFixedPeriodQuestionnaire(evakaClock.today())) } }
+        return db.connect { dbc ->
+            dbc.read { tx ->
+                val activeQuestionnaire = tx.getActiveFixedPeriodQuestionnaire(evakaClock.today())
+                if (activeQuestionnaire == null) listOf()
+                else {
```
Comment (kimifoud) [inline service/src/main/kotlin/fi/espoo/evaka/holidayperiod/HolidayPeriodControllerCitizen.kt:61]: nitpicking: nesting with else is redundant and less readable

File: frontend/src/lib-customizations/espoo/citizen/assets/i18n/sv.tsx [hunk 2]
```diff
@@ -271,7 +272,9 @@ const sv: Translations = {
       holidayFor: 'Semestertid för:',
       childOnHoliday: 'Barnet är på semester',
       addTimePeriod: 'Lägg till en period',
-      emptySelection: 'Ingen gratis ledighet'
+      emptySelection: 'Ingen gratis ledighet',
+      notEligible: (period: FiniteDateRange) =>
+        `Enkäten rör inte barnet, eftersom hen inte varit i småbarnspedagogiken oavbrutet under tiden ${period.start.format()} - ${period.end.format()}.`
```
Comment (kimifoud) [inline frontend/src/lib-customizations/espoo/citizen/assets/i18n/sv.tsx:277]: `period.formatCompact()` maybe?

File: frontend/src/employee-frontend/components/holiday-periods/FixedPeriodQuestionnaireForm.tsx [hunk 3]
```diff
@@ -484,6 +549,47 @@ export default React.memo(function FixedPeriodQuestionnaireForm({
...
           hideErrorsBeforeTouched={hideErrorsBeforeTouched}
         />
+
+        <Label>{i18n.holidayQuestionnaires.conditionContinuousPlacement}</Label>
+        <FixedSpaceRow alignItems="center">
+          <DatePicker
+            info={useMemo(
+              () => errorToInputInfo(errors.start, i18n.validationErrors),
+              [errors.start, i18n.validationErrors]
+            )}
```
Comment (kimifoud) [inline frontend/src/employee-frontend/components/holiday-periods/FixedPeriodQuestionnaireForm.tsx:559]: Both of error conditions are wrong (`errors.start` and `errors.end`)

**Review by kimifoud — CHANGES_REQUESTED** [review]
Looks good! Some minor observations.

File: service/src/main/kotlin/fi/espoo/evaka/holidayperiod/HolidayPeriodControllerCitizen.kt (hunk 1 above)
Comment (akheron) [reply service/src/main/kotlin/fi/espoo/evaka/holidayperiod/HolidayPeriodControllerCitizen.kt]: 👍

File: frontend/src/lib-customizations/espoo/citizen/assets/i18n/sv.tsx (hunk 2 above)
Comment (akheron) [reply frontend/src/lib-customizations/espoo/citizen/assets/i18n/sv.tsx]: Didn't know of such a function! 😄 Changed to `period.format()` in this case.

File: frontend/src/employee-frontend/components/holiday-periods/FixedPeriodQuestionnaireForm.tsx (hunk 3 above)
Comment (akheron) [reply frontend/src/employee-frontend/components/holiday-periods/FixedPeriodQuestionnaireForm.tsx]: Fixed 👍

**Review by kimifoud — APPROVED** [review]
💎

**Pull request merged**